### PR TITLE
Support remote logging

### PR DESCRIPTION
**Pull request by seiyanuta**

### CODE REVIEW

File: app/models/device_status.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    DEVICE_STATUSES.include?(@status)
+  end
+
+  def save
+    unless validate
+      return false
+    end
+
+    Redis.current.set(@key, @status)
+    true # TODO: error handling
```
Comment (seiyanuta) [inline app/models/device_status.rb]: I'll address this and push changes soon.

Comment (seiyanuta) [inline app/models/device_status.rb]: I've done this.

Comment (seiyanuta) [inline app/models/device_status.rb]: https://github.com/codestand/mgmt/pull/15/commits/23abad1dd2039858a26dd9b3c6de144a9cd375f4

**Comment by seiyanuta** [comment]
Fixed CI builds and added an endpoint to download log messages.

File: app/controllers/baseos_controller.rb [hunk 2]
```diff
@@ -2,9 +2,17 @@ class BaseosController < ApplicationController
...
-    @device.save!
+    device_secret = @device.device_secret
+    heartbeat = Heartbeat.new(device_secret: device_secret)
+    device_status = DeviceStatus.new(device_secret: device_secret,
+                                     status: heartbeat_params[:status])
+    logging = Logging.new(device_secret: device_secret,
+                          lines: request.raw_post.split("\n"))
+
+    unless [heartbeat.save, device_status.save, logging.save].all?
+      logging.warn "failed to save to Redis"
```
Comment (sh19910711) [inline app/controllers/baseos_controller.rb]: `logger`?

File: app/models/heartbeat.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Heartbeat
+  def initialize(device_secret)
+    @key = "devices:hearbeat:#{device_secret}"
```
Comment (sh19910711) [inline app/models/heartbeat.rb]: ty(p)o

**Comment by sh19910711** [comment]
LGTM in the other parts 👍

File: app/models/heartbeat.rb (hunk 3 above)
Comment (seiyanuta) [inline app/models/heartbeat.rb]: I'll fix it. What does it mean `(p)` by the way?

File: app/controllers/baseos_controller.rb (hunk 2 above)
Comment (seiyanuta) [inline app/controllers/baseos_controller.rb]: fixed

File: app/models/heartbeat.rb (hunk 3 above)
Comment (seiyanuta) [inline app/models/heartbeat.rb]: fixed

Comment (sh19910711) [inline app/models/heartbeat.rb]: hear(t)beat

**Comment by sh19910711** [comment]
Seems good :+1: Thank you.

**Pull request merged**